src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2349,10 +2345,7 @@ impl Coordinator {
 
         // Determine the frontier of updates to subscribe *from*.
         // Updates greater or equal to this frontier will be produced.
-        let uses = match from {
-            SubscribeFrom::Id(id) => vec![id],
-            SubscribeFrom::Query { .. } => depends_on,
-        };
+        let uses = from.depends_on();
```

Comment:
We end up saving this inside the `ActiveSubscribe` struct on line 2435. We use this set to determine if a `Subscribe` should be cancelled when one of it's dependencies is dropped. It feels like this maybe should actually be the set of resolved IDs, though I'm not sure. What do you think?

Reply:
Great callout. I'm not sure. I _think_ using the expression dependencies here is right? Imagine you ran `SUBSCRIBE (SHOW INDEXES FROM <table>)`, for example. I think the right thing to do there is error if (somehow) `mz_indexes` got dropped. But you don't need to error (I think?) if `<table>` gets dropped—you just need to produce a retraction of all the dropped indexes.

Reply:
I'm a bit torn to be honest. `SHOW INDEXES FROM <table>` does return an error if `<table>` is not a valid object, so it seems reasonable that an in progress subscribe should stop when `<table>` stops becoming a valid object.

On the other hand, if you have a scenario like this:
```
CREATE TABLE t1 (a TEXT);
CREATE TABLE t2 (b TEXT);

SUBSCRIBE (SELECT * FROM t1 LEFT JOIN  (SHOW INDEXES FROM t2) AS s ON t1.a = s.name)
```

Then the subscribe can keep emitting valid diffs after `t2` has been dropped.

I think this is more of a UX concern than a correctness concern. Using just the expression dependencies is fine from a correctness perspective, nothing will blow up or panic from that. So we can probably merge as is and resolve this in a later PR.

Reply:
👍🏽 sounds good to me.